oak-search/src/test/java/org/apache/jackrabbit/oak/plugins/index/FullTextIndexCommonTest.java

```diff
@@ -112,6 +113,67 @@ public void testFullTextTermWithUnescapedBraces() throws Exception {
+
+        root.commit();
+
+        assertEventually(() -> {
+            assertQuery("//*[j:c/@analyzed_field = 'bar']", XPATH, Arrays.asList("/test/a", "/test/c/d"));
+            assertQuery("//*[d/*/@analyzed_field = 'bar']", XPATH, Collections.singletonList("/test/c"));
+        });
+    }
+
+    @Test
+    public void pathTransformationsWithPathRestrictions() throws Exception {
+        Tree test = setup();
+
+        test.addChild("a").addChild("j:c").setProperty("analyzed_field", "bar");
+        test.addChild("b").setProperty("analyzed_field", "bar");
+        test.addChild("c").addChild("d").addChild("j:c").setProperty("analyzed_field", "bar");
+        test.addChild("e").addChild("temp:c").setProperty("analyzed_field", "bar");
+        test.addChild("f").addChild("d").addChild("temp:c").setProperty("analyzed_field", "bar");
+        test.addChild("g").addChild("e").addChild("temp:c").setProperty("analyzed_field", "bar");
+
+
+        Tree temp = root.getTree("/").addChild("tmp");
+
+        temp.addChild("a").addChild("j:c").setProperty("analyzed_field", "bar");
+        temp.addChild("b").setProperty("analyzed_field", "bar");
+        temp.addChild("c").addChild("d").addChild("j:c").setProperty("analyzed_field", "bar");
+        root.commit();
+
+        assertEventually(() -> {
+            // ALL CHILDREN
+            assertQuery("/jcr:root/test//*[j:c/analyzed_field = 'bar']", XPATH, Arrays.asList("/test/a", "/test/c/d"));
+            assertQuery("/jcr:root/test//*[*/analyzed_field = 'bar']", XPATH, Arrays.asList("/test/a", "/test/c/d", "/test/e", "/test/f/d", "/test/g/e"));
+            assertQuery("/jcr:root/test//*[d/*/analyzed_field = 'bar']", XPATH, Arrays.asList("/test/c", "/test/f"));
+            assertQuery("/jcr:root/test//*[analyzed_field = 'bar']", XPATH, Arrays.asList("/test/a/j:c","/test/b","/test/c/d/j:c",
+                    "/test/e/temp:c", "/test/f/d/temp:c","/test/g/e/temp:c"));
+
+            // DIRECT CHILDREN
+            assertQuery("/jcr:root/test/*[j:c/analyzed_field = 'bar']", XPATH, Arrays.asList("/test/a"));
+            assertQuery("/jcr:root/test/*[*/analyzed_field = 'bar']", XPATH, Arrays.asList("/test/a", "/test/e"));
+            assertQuery("/jcr:root/test/*[d/*/analyzed_field = 'bar']", XPATH, Arrays.asList("/test/c", "/test/f"));
+            assertQuery("/jcr:root/test/*[analyzed_field = 'bar']", XPATH, Arrays.asList("/test/b"));
+
+            // EXACT
+            assertQuery("/jcr:root/test/a[j:c/analyzed_field = 'bar']", XPATH, Arrays.asList("/test/a"));
+            assertQuery("/jcr:root/test/a[*/analyzed_field = 'bar']", XPATH, Arrays.asList("/test/a"));
+            assertQuery("/jcr:root/test/c[d/*/analyzed_field = 'bar']", XPATH, Arrays.asList("/test/c"));
+            assertQuery("/jcr:root/test/a/j:c[analyzed_field = 'bar']", XPATH, Arrays.asList("/test/a/j:c"));
+
+            // TODO : We don't have assertions for PARENT path filter here.
```

Comment:
Parent path filters are (sometimes) used for joins, depending on the join order... I think we would need to use SQL-2 queries; not sure if it is better to test it? If you think it's useful, I can have a look...

Reply:
@thomasmueller  - yes I think it would be better to test it. But I was not able to figure out the query that might be useful to validate this use case. Would appreciate if you can help out with the query - I will modify the test case to include that scenario as well.

Reply:
@thomasmueller  thanks for the help with query. I added the new test.